pkg/apis/policy/validation/validation.go

```diff
@@ -50,8 +50,20 @@ func ValidatePodDisruptionBudgetUpdate(pdb, oldPdb *policy.PodDisruptionBudget)
 func ValidatePodDisruptionBudgetSpec(spec policy.PodDisruptionBudgetSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allErrs = append(allErrs, extensionsvalidation.ValidatePositiveIntOrPercent(spec.MinAvailable, fldPath.Child("minAvailable"))...)
-	allErrs = append(allErrs, extensionsvalidation.IsNotMoreThan100Percent(spec.MinAvailable, fldPath.Child("minAvailable"))...)
+	if spec.MinAvailable != nil && spec.MaxUnavailable != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, spec, "MinAvailable and MaxUnavailable cannot be both set"))
+	}
+
```

Comment:
There are multiple ways to deal with this situation IMO, and it would be driven by policy. The admin could either set an SLA and force-drain the node after a certain timeout, or reach out to the application owner. One can imagine having an admission controller that prevents it in the first place. I think we should defer the decision on handling these pathological cases to admission control and keep it out of the core object's validation. @smarterclayton @davidopp @erictune does that make sense?